src/store/user.ts

```diff
@@ -7,34 +7,24 @@ export const useUserStore = defineStore('user', {
   state: () => {
     return {
       localeOptions: process.env.VUE_APP_LOCALES ? JSON.parse(process.env.VUE_APP_LOCALES) : { "en": "English" },
-      preference: {
-        locale: 'en'
-      } as any
+      locale: 'en'
     }
   },
   getters: {
-    getLocale: (state) => state.preference.locale,
+    getLocale: (state) => state.locale,
     getLocaleOptions: (state) => state.localeOptions
   },
   actions: {
     setLocale(payload: string) {
       // update locale in state and globally
       i18n.global.locale.value = payload
-      this.setPreference({ locale: payload })
+      this.setPreference(payload)
```

Comment:
Don't define a separate method for updating locale, as here we are defining separate method just for api call, and that method will not be used separately and will always be used with this method.